pytest_client_tools/config.py

```diff
@@ -0,0 +1,32 @@
+# SPDX-FileCopyrightText: Red Hat
+# SPDX-License-Identifier: MIT
+
+import pytest
+from dynaconf import Dynaconf, Validator
+
+#
+# it is necessary to use define global variable since dynaconf settings object
+# is used even in pytest hooks to get config properties
+#
+_settings = Dynaconf(
+    envvar_prefix="PYTEST_CLIENT_TOOLS",
+    settings_files=["settings.toml", ".secrets.toml"],
+)
+
+
+@pytest.fixture
+def settings(scope="session"):
+    _settings.validators.register(
+        Validator("candlepin.host"),
+        Validator("candlepin.port", gt=0, lt=65536, cast=int, is_type_of=int),
+        Validator("candlepin.prefix", startswith="/"),
+        Validator("candlepin.insecure", cast=bool, is_type_of=bool),
+        Validator("candlepin.username"),
+        Validator(
+            "candlepin.password",
+            must_exist=True,
+            when=Validator("candlepin.username", must_exist=True),
+        ),
+    )
+    _settings.validators.validate()
+    return _settings
```

Comment:
The main reason for this simplification was that one class in a module is redundant in case test_config. It seems to me that a module file is enough to group functions into one namespace. I vote for keeping it super simple. ie. no hidden logic for a tester, upstream visitor to learn. That is a reason to use only well known tools, libraries rather than creating our own classes with another middleware.

Reply:
No, `TestConfig` is not redundant, and it does not create any new middleware. As I said, its purpose is to group anything related to the configuration of the test session, i.e. what is read using dynaconf plus additional logic related to that.